test/source/libs/gen_allocation.py

```diff
@@ -0,0 +1,168 @@
+from optparse import OptionParser
+parser = OptionParser()
+parser.add_option('-t', '--tests', dest='test_filename', help='Write test cases to FILE', metavar='FILE')
+parser.add_option('-f', '--funcs', dest='func_filename', help='Write test functions to FILE', metavar='FILE')
+(options, args) = parser.parse_args()
+
+if options.test_filename is None:
+    parser.print_help()
+    exit()
+if options.func_filename is None:
+    parser.print_help()
+    exit()
+
+# ------------------------------------------ #
+
+types = [
+    {
+        'category': "Integral Types",
+        'types': [
```

Comment:
Ah this could serve as a nice lookup - we could convert this (and other similar tests) to docs when all is said and done. It's nice to see them in one place.

Reply:
Yeah, definitely. I felt it was a good start at trying to separate out what we want to test versus how we implement it. Some testing docs will definitely be in our future!